src/twisted/internet/defer.py

```diff
@@ -159,6 +165,65 @@ def maybeDeferred(f, *args, **kw):
         return succeed(result)
 
 
+def unbox(deferred, raises=True, passthrough=True):
```

Comment:
After thinking about this more, I *think* the right answer is one of:

* Always pass through successful responses and always consume errors.
* Always pass through successful responses and add a ``consumeErrors`` parameter.
  * Defaulting to ``True`` to do the right thing by default.
  * Defaulting to ``False`` to match ``gatherResults`` and friends.

The reasoning for this is that there's very little downside to passing through a successful result. At worst it will just hold a reference to it until the ``Deferred`` itself gets garbage collected. So it just always passes that through.

However, if you don't consume errors it prints spurious error results to stdout so there is an incentive to consume those and not pass them through. Adding a ``consumeErrors`` parameter aligns with other utilities and allows control on this.

The downside is errors and success ends up being treated differently although that could possibly be solved by consuming everything and treating ``unbox()`` as something you should generally only call on a ``Deferred`` that is no longer going to be kept around or used.